Build failure: libgda-5.2.10

## Steps To Reproduce

Steps to reproduce the behavior:

1. build libgda-5.2.10

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

https://gist.github.com/ushitora-anqou/96dd3eec4de775dc07c710ba40cd0d52

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.66, NixOS, 25.05 (Warbler), 25.05.20241219.d70bd19`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixos-24.05"`
 - nixpkgs: `/nix/store/8fwsiv0hd7nw1brkvka0jf1frk3m7qkr-source`


## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

I have the same problem 


`error: builder for '/nix/store/srn63gzxdrrw52pxba2dgd22f1pl8iv1-libgda-5.2.10.drv' failed with exit code 2;
       last 25 log lines:
       >    77 |     (g_string_free) ((str), (free_segment)))
       >       |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > gda-data-model-dir.c:547:9: note: in expansion of macro 'g_string_free'
       >   547 |         g_string_free (md5str, FALSE);
       >       |         ^~~~~~~~~~~~~
       > In file included from /nix/store/y8fxzs8srzd6d74a85zbpssjr8wkj9q4-glib-2.82.1-dev/include/glib-2.0/glib/glist.h:34,
       >                  from /nix/store/y8fxzs8srzd6d74a85zbpssjr8wkj9q4-glib-2.82.1-dev/include/glib-2.0/glib/ghash.h:36,
       >                  from /nix/store/y8fxzs8srzd6d74a85zbpssjr8wkj9q4-glib-2.82.1-dev/include/glib-2.0/glib.h:52:
       > gda-data-model.c: In function 'gda_data_model_import_from_model':
       > /nix/store/y8fxzs8srzd6d74a85zbpssjr8wkj9q4-glib-2.82.1-dev/include/glib-2.0/glib/gmem.h:270:19: warning: argument 1 range [18446744071562067968, 18446744073709551615] exceeds maximum object size 9223372036854775807 [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Walloc-size-larger-than_003d-Walloc-size-larger-than=8;;]
       >   270 |             __p = g_##func##_n (__n, __s);                      \
       >       |                   ^~~~~~~~~~~~~~~~~~~~~~~
       > /nix/store/y8fxzs8srzd6d74a85zbpssjr8wkj9q4-glib-2.82.1-dev/include/glib-2.0/glib/gmem.h:332:57: note: in expansion of macro '_G_NEW'
       >   332 | #define g_new0(struct_type, n_structs)                  _G_NEW (struct_type, n_structs, malloc0)
       >       |                                                         ^~~~~~
       > gda-data-model.c:2180:24: note: in expansion of macro 'g_new0'
       >  2180 |         append_types = g_new0 (GType, to_nb_cols);
       >       |                        ^~~~~~
       > /nix/store/y8fxzs8srzd6d74a85zbpssjr8wkj9q4-glib-2.82.1-dev/include/glib-2.0/glib/gmem.h:102:10: note: in a call to allocation function 'g_malloc0_n' declared here
       >   102 | gpointer g_malloc0_n      (gsize         n_blocks,
       >       |          ^~~~~~~~~~~
       > make[2]: Leaving directory '/build/libgda-5.2.10/libgda'
       > make[1]: *** [Makefile:1275: all-recursive] Error 1
       > make[1]: Leaving directory '/build/libgda-5.2.10/libgda'
       > make: *** [Makefile:742: all-recursive] Error 1
       For full logs, run 'nix log /nix/store/srn63gzxdrrw52pxba2dgd22f1pl8iv1-libgda-5.2.10.drv'.
error: 1 dependencies of derivation '/nix/store/rz984649ycpd10dj6jb7p2jc34wlcx8n-gnome-shell-extension-pano-v23-alpha3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/0d508dxff9dlmm4kijhvjq2631rv0hnr-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/bw0hpdf6drxxqfc7rkv7f1gblg8ymzj5-nixos-system-nixos-25.05.20241227.634fd46.drv' failed to build`

I'm tagging the pkg maintainers:
@bobby285271 @hedning @jtojnar @dasj19 of the pkg
@honnip of the extension that failed to build because of the dependencie

--- a/pkgs/development/libraries/libgda/default.nix
+++ b/pkgs/development/libraries/libgda/default.nix
@@ -86,7 +86,7 @@ stdenv.mkDerivation rec {
     "--enable-system-sqlite=${if stdenv.hostPlatform.isDarwin then "no" else "yes"}"
   ];
 
-  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang "-Wno-error=incompatible-function-pointer-types";
+  env.NIX_CFLAGS_COMPILE = "-Wno-error=incompatible-pointer-types";
 
   enableParallelBuilding = true;
 

